File: 3876-construct-uniform-parity-array-ii/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    bool canConstructArray(std::vector<int>& nums1) {
        int min_val = nums1[0];
        bool all_even = true;

        for (int x : nums1) {
            if (x < min_val) {
                min_val = x;
            }
            if (x % 2 != 0) {
                all_even = false;
            }
        }

        // We can succeed if the minimum element is odd (making all elements odd)
        // or if all elements are already even (making all elements even).
        return (min_val % 2 != 0) || all_even;
    }
};

File: 3876-construct-uniform-parity-array-ii/solution.py
class Solution:
    def canConstructArray(self, nums1: list[int]) -> bool:
        min_val = min(nums1)
        all_even = all(x % 2 == 0 for x in nums1)
        
        # We can succeed if the minimum element is odd (making all elements odd)
        # or if all elements are already even (making all elements even).
        return (min_val % 2 != 0) or all_even